File: backend/models/entities/transaction.js
import { Model } from 'sequelize'

export default (sequelize, DataTypes) => {
  class Transaction extends Model {
    static associate(models) {
      // Transaction belongs to User
      Transaction.belongsTo(models.User, {
        foreignKey: 'userId',
        as: 'user',
      })
    }
  }

  Transaction.init(
    {
      transactionId: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
        allowNull: false,
      },
      userId: DataTypes.INTEGER,
      money: DataTypes.INTEGER,
      type: DataTypes.STRING,
      time: DataTypes.DATE,
      status: DataTypes.STRING,
      finishAt: DataTypes.DATE,
    },
    {
      sequelize,
      modelName: 'Transaction',
      tableName: 'transaction',
      timestamps: false,
    }
  )

  return Transaction
}

File: frontend/app/src/main/java/com/example/frontend/data/model/Transaction.kt
package com.example.frontend.data.model

import java.time.LocalDateTime

data class Transaction (
    val transactionId: Int,
    val user: User?=null,
    val money: Int=0,
    val type: String,
    val time: LocalDateTime,
    val status: String,
    val finishAt: LocalDateTime?=null
)

data class userTransaction (
    val userId : Int,
    val userName : String,
    val dUserName : String,
    val avatarId : String
)

data class Transaction2 (
    val transactionId: Int,
    val userId: Int,
    val money: Int,
    val type: String,
    val time: String,
    val status: String,
    val finishAt: String,
    val user : userTransaction
)